hrms/hr/report/shift_attendance/shift_attendance.js

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2023, Frappe Technologies Pvt. Ltd. and contributors
+// For license information, please see license.txt
+
+frappe.query_reports["Shift Attendance"] = {
+  filters: [
+    {
+      fieldname: "from_date",
+      label: __("From Date"),
+      fieldtype: "Date",
+    },
+    {
+      fieldname: "to_date",
+      label: __("To Date"),
+      fieldtype: "Date",
+    },
+    {
+      fieldname: "employee",
+      label: __("Employee"),
+      fieldtype: "Link",
+      options: "Employee",
+    },
+    {
+      fieldname: "shift",
+      label: __("Shift Type"),
+      fieldtype: "Link",
+      options: "Shift Type",
+    },
+    {
+      fieldname: "department",
+      label: __("Department"),
+      fieldtype: "Link",
+      options: "Department",
+    },
+    {
+      fieldname: "company",
+      label: __("Company"),
+      fieldtype: "Link",
+      options: "Company",
+    },
+  ],
+  formatter: (value, row, column, data, default_formatter) => {
+    value = default_formatter(value, row, column, data);
```

Comment:
We can also show early entries and late exits in green to make it more informative. Basically rows with IN < Start Time / OUT > End Time. What do you think?